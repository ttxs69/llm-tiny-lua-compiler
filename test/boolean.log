Advanced to token TOKEN_PRINT 'print'
Advanced to token TOKEN_LPAREN '('
Advanced to token TOKEN_TRUE 'true'
Advanced to token TOKEN_RPAREN ')'
Advanced to token TOKEN_PRINT 'print'
Advanced to token TOKEN_LPAREN '('
Advanced to token TOKEN_FALSE 'false'
Advanced to token TOKEN_RPAREN ')'
Advanced to token TOKEN_PRINT 'print'
Advanced to token TOKEN_LPAREN '('
Advanced to token TOKEN_NIL 'nil'
Advanced to token TOKEN_RPAREN ')'
Advanced to token TOKEN_IF 'if'
Advanced to token TOKEN_TRUE 'true'
Advanced to token TOKEN_THEN 'then'
Advanced to token TOKEN_PRINT 'print'
Advanced to token TOKEN_LPAREN '('
Advanced to token TOKEN_STRING '"true is true"'
Advanced to token TOKEN_RPAREN ')'
Advanced to token TOKEN_END 'end'
Advanced to token TOKEN_IF 'if'
Advanced to token TOKEN_FALSE 'false'
Advanced to token TOKEN_THEN 'then'
Advanced to token TOKEN_PRINT 'print'
Advanced to token TOKEN_LPAREN '('
Advanced to token TOKEN_STRING '"false is true"'
Advanced to token TOKEN_RPAREN ')'
Advanced to token TOKEN_ELSE 'else'
Advanced to token TOKEN_PRINT 'print'
Advanced to token TOKEN_LPAREN '('
Advanced to token TOKEN_STRING '"false is false"'
Advanced to token TOKEN_RPAREN ')'
Advanced to token TOKEN_END 'end'
Advanced to token TOKEN_IF 'if'
Advanced to token TOKEN_NIL 'nil'
Advanced to token TOKEN_THEN 'then'
Advanced to token TOKEN_PRINT 'print'
Advanced to token TOKEN_LPAREN '('
Advanced to token TOKEN_STRING '"nil is true"'
Advanced to token TOKEN_RPAREN ')'
Advanced to token TOKEN_ELSE 'else'
Advanced to token TOKEN_PRINT 'print'
Advanced to token TOKEN_LPAREN '('
Advanced to token TOKEN_STRING '"nil is false"'
Advanced to token TOKEN_RPAREN ')'
Advanced to token TOKEN_END 'end'
Advanced to token TOKEN_EOF ''
