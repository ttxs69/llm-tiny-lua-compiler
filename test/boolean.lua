print(true)
print(false)
print(nil)

if true then
  print("true is true")
end

if false then
  print("false is true")
else
  print("false is false")
end

if nil then
  print("nil is true")
else
  print("nil is false")
end