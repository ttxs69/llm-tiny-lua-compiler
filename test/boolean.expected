true
false
nil
true is true
false is false
nil is false
